### PR TITLE
Fix h3 timing stats (+tests)

### PR DESCRIPTION
**Pull request by gfx**
Added a test file to make sure HTTP/3 timing stats are expected. 

This test depends on https://github.com/h2o/h2o/pull/2973, with which I've confirmed the test passes.

### CODE REVIEW

File: t/50logconf.t [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+use File::Temp qw(tempdir);
+use JSON;
+use Time::HiRes qw(sleep);
+use Test::More;
+use t::Util;
+
+my $client_prog = bindir() . "/h2o-httpclient";
+plan skip_all => "$client_prog not found"
+    unless -e $client_prog;
+plan skip_all => 'mruby support is off'
```
Comment (sharksforarms) [inline t/50logconf.t:15]: is ruby support required? I was able to run this test without it

**Comment by sharksforarms** [comment]
Thanks for creating this test!

File: t/50logconf.t (hunk 1 above)
Comment (gfx) [reply t/50logconf.t]: Oops, removed it! Thanks.

Feel free to merge this PR to yours, https://github.com/h2o/h2o/pull/2973.

**Comment by sharksforarms** [comment]
Integrated into https://github.com/h2o/h2o/pull/2973

**Review by kazuho — COMMENTED** [review]
Thank you for the PR. Yes, we need to test these.

The code looks good modulo two points:
* The file name better be changed. logconf is a module that builds logs, not only these timing metrics.
* I'm not sure if we want to write test for each protocol version. HTTP-level metrics are HTTP-version-independent, so should be the tests. If there has to be differences between the versions, it would be more natural to code them as exceptions.

**Comment by gfx** [comment]
@kazuho 

Thanks for your comments. I'm focusing only on timing metrics to the file right now, but I'll add more metrics to it later. So the timing metrics testing can be refactored to a single subroutine, but I'd like to keep the filename and the entire structure of the test file.

**Comment by kazuho** [comment]
@gfx That observation I disagree with.

We already have tests for various logging directives in t/50access-log.t. Therefore, if the intent is to have tests for various logging directives, tests should be added to that file.

Or else if the intent is to have tests for server-timings (I think such intent is not unreasonable, as server-timings are bit special in sense that they use trailers), maybe we should _also_ be adding tests to existing ones; t/50server-timing.t, t/50reverse-proxy-timings.t.

**Review by kazuho — APPROVED** [review]
@sharksforarms @gfx Thank you for all your efforts. Everything looks good, to be merged once CI succeeds.

Regarding tests, I like how @gfx has refactored the existing one. I see improvements in how Test::More and regular expressions are used.

**Pull request merged**